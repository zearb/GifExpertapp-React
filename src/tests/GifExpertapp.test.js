import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import GifExpertApp from '../GifExpertapp';

describe('Pruebas en <GifExpertapp/>',()=>{


    test('Se debe mostrar correctamente',()=>{

        const wrapper = shallow(<GifExpertApp />);

        expect(wrapper).toMatchSnapshot();

    });

    test('Debe mostrar una lista de categorÃ­as',()=>{

        const categories = ['DBZ','Pizza'];
        const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);

        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('GifGrid').length).toBe(categories.length);

    });


});