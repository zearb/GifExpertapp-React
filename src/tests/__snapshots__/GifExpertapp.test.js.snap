// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertapp/> Debe mostrar una lista de categor√≠as 1`] = `
<Fragment>
  <h1
    className="animate__animated animate__fadeInDown"
  >
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <GifGrid
    category="DBZ"
    key="DBZ"
  />
  <GifGrid
    category="Pizza"
    key="Pizza"
  />
</Fragment>
`;

exports[`Pruebas en <GifExpertapp/> Se debe mostrar correctamente 1`] = `
<Fragment>
  <h1
    className="animate__animated animate__fadeInDown"
  >
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
</Fragment>
`;
